File: src/Java/CreationalPatterns/AbstractFactory/EraStyles/src/ConcreteFactories/ShipFactory.java
package ConcreteFactories;

import Abstraction.EraObject;
import Abstraction.EraObjectStylesFactory;
import Products.MedievalShip;
import Products.RenaissanceShip;
import Products.VictorianShip;

public class ShipFactory extends EraObjectStylesFactory {
    @Override
    public EraObject createMedievalStyleObject() {
        return new MedievalShip();
    }

    @Override
    public EraObject createRenaissanceStyleObject() {
        return new RenaissanceShip();
    }

    @Override
    public EraObject createVictorianEraStyleObject() {
        return new VictorianShip();
    }
}

File: src/C#/CreationalPatterns/AbstractFactory/EraStyles/ConcreteFactories/ShipFactory.cs
using EraStyles.Abstraction;
using EraStyles.Products;
using System;
using System.Collections.Generic;
using System.Text;

namespace EraStyles.ConcreteFactories
{
    public class ShipFactory : EraObjectStylesFactory
    {
        public override EraObject CreateMedievalStyleObject()
        {
            return new MedievalShip();
        }

        public override EraObject CreateRenaissanceStyleObject()
        {
            return new RenaissanceShip();
        }

        public override EraObject CreateVictorianEraStyleObject()
        {
            return new VictorianShip();
        }
    }
}
